dvl-a50/dvl.py

```diff
@@ -175,12 +194,44 @@ def set_should_send(self, should_send):
         self.should_send = should_send
         self.save_settings()
 
+    def longitude_scale(self, lat):
+        scale = math.cos(lat * (math.pi / 180.0))
+        return max(scale, 0.01)
+
+    def latLngToXY (self, lat, lon):
+        x = (lat-self.origin[0]) * LATLON_TO_CM
+        y =  self.longitude_scale((lat + self.origin[0])/2) * LATLON_TO_CM * (lon-self.origin[1])
+        return [x, y]
+
+    def has_origin_set(self):
+        data = json.loads(self.mav.get("/GLOBAL_POSITION_INT/message"))
+        return data['lat'] != 0 or data['lon'] != 0
```

Comment:
Perhaps worth a shot?
```suggestion
    def has_origin_set(self):
        # Check for record of an origin confirmation message from the autopilot
        return bool(self.mav.get("/GPS_GLOBAL_ORIGIN/message"))
```

Reply:
The `GPS_GLOBAL_ORIGIN` approach requires also checking that the message was received after the last autopilot restart (assuming that's even possible to check for), otherwise restarting the autopilot would require manually restarting the DVL service (or mavlink2rest).

If the `GLOBAL_POSITION_INT` approach works the way you expected then it's a more robust approach, because it would only have issues right after the autopilot restart (i.e. setting a new 'current location' wouldn't work properly because there's no GPS origin set, but the previous non-zero global position is still in mavlink2rest), but the functionality would then start working again as soon as a new `GLOBAL_POSITION_INT` message is received from the vehicle, so it's self-correcting within 1/3 of a second, which is likely fine.